src/Engine/EmitMSIL/Replication.cs

```diff
@@ -220,15 +220,13 @@ public static IList ReplicationLogic(IEnumerable<MethodBase> mInfos, IList args,
             // TODO_MSIL: Emit these CLRStackValue's from the CodeGen stage.
             var stackValues = MarshalFunctionArguments(args, runtimeCore);
 
-            var reducedArgs = ReduceArgs(stackValues);
```

Comment:
`ReduceArgs` would assume that if there is an argument that is an `IList` of a single value, it would need to be reduced to a single value. Firstly, I believe this assumption is flawed, secondly, I don't believe it's required anymore given that we are not wrapping single values as an `IList `of a single value anymore.